dev/tasks/tasks.yml

```diff
@@ -428,7 +428,7 @@ tasks:
       python_version: "{{ python_version }}"
       python_abi_tag: "{{ abi_tag }}"
       macos_deployment_target: "{{ macos_version }}"
-      runs_on: "macos-12"
+      runs_on: "macos-13"
```

Comment:
Should we keep the deployment target at 12.0? 
Because right now this PR is not just a "CI" PR, it affects Python packaging, making our wheels no longer install on MacOS 12

Reply:
Ah, should we keep macOS 12 support for wheels?
Can we continue to discuss this in this thread https://lists.apache.org/thread/03dqppm1kc26nfppk8wcj0tnzh1st3mx ?